pool-buffer.c

```diff
@@ -120,7 +122,11 @@ void finish_buffer(struct pool_buffer *buffer) {
 }
 
 struct pool_buffer *get_next_buffer(struct wl_shm *shm,
-		struct pool_buffer pool[static 2], uint32_t width, uint32_t height) {
+		struct pool_buffer pool[static 2], uint32_t width, uint32_t height, double scale) {
```

Comment:
I'd rather not pass down the scale to `get_next_buffer`. `get_next_buffer` doesn't need to care about scale. Instead we can just reset the scale right before rendering:

```c
cairo_identity_matrix(cairo);
cairo_scale(cairo, scale, scale);
```